Add Highlight postprocessing to Editor selection (remove bouncing animation)

DasCapschen opened the pull request:
This removes the "bouncing" animation from the selected object in Editor, and replaces it with Postprocessing Effects to outline and highlight the object.

This is what it would currently look like:
![image](https://user-images.githubusercontent.com/3923406/139576736-3afc757a-d5fb-4091-9948-22ee2f4ac404.png)

Of course, we can decide to only use outlines, or only use highlighting (they are 2 separate nodes that added to the Editor scene tree), and the colors are also configurable.

What is currently not configurable, though I'm sure I can work out how to do that, is the thickness of the outline, which is currently only about 2 pixels...

HaSa1002 commented:
I'd be in favor of only using the outline shader because the tint makes it hard to match colors.

HaSa1002 commented on src/addons/Libre_Train_Sim_Editor/Data/Modules/Postprocess/LaplaceEdgeDetect.gdshader:
```diff
@@ -0,0 +1,38 @@
+// which means only black or white, depending on if a pixel contains information
+// or not. 
+float tex_bin(sampler2D tex, vec2 uv) {
+	vec3 rgb = texture(tex, uv).xyz;
+	if((rgb) == vec3(0)) {
+		return 0.0;
+	} else {
+		return 1.0;
+	}
```
`step`?

DasCapschen replied:
what?

HaSa1002 replied:
```suggestion
	return step(texture(tex, uv).xyz, vec3(0)).x;
```

HaSa1002 replied:
because step is GPU accelerated it is faster (in theory)

DasCapschen replied:
step is not going to work. It is defined as `b < a ? 0 : 1` and the minimum value rgb can have *is* 0, so `0 < 0` is always false and this will always return 1...

HaSa1002 replied:
yes. you're right

HaSa1002 commented on src/addons/Libre_Train_Sim_Editor/Data/Modules/Postprocess/ShaderPP.gd:
```diff
@@ -0,0 +1,31 @@
+
+export var highlight_color: Color
+
+var external_camera
+var shader_camera
+
+func _ready():
+	external_camera = get_viewport().get_camera()
+	shader_camera = find_node("Camera")
```
```suggestion
onready var external_camera := get_viewport().get_camera()
onready var shader_camera := find_node("Camera") as Camera
```

HaSa1002 commented on src/addons/Libre_Train_Sim_Editor/Data/Modules/Postprocess/ShaderPP.gd:
```diff
@@ -0,0 +1,31 @@
+	external_camera = get_viewport().get_camera()
+	shader_camera = find_node("Camera")
+
+func _process(delta: float) -> void:
```
```suggestion
func _process(_delta: float) -> void:
```
Surpresses unused argument warning

HaSa1002 commented on src/addons/Libre_Train_Sim_Editor/Data/Modules/Postprocess/highlight_shader.gdshader:
```diff
@@ -0,0 +1,23 @@
+// or not. 
+int tex_bin(sampler2D tex, vec2 uv) {
+	vec3 rgb = texture(tex, uv).xyz;
+	if((rgb) == vec3(0)) {
+		return 0;
+	} else {
+		return 1;
+	}
```
`step`?

DasCapschen commented:
I've changed it to only use outlines. 
And I fixed the outline shader... it was discarding half of the outline 🙈 
So they are bigger now too, should be well visible.

DasCapschen commented:
rebased

HaSa1002 reviewed (approved):
I mean it is ugly, but who cares xD
![grafik](https://user-images.githubusercontent.com/14185889/140583227-5578f105-1609-491b-8c35-565a971e4b94.png)

HaSa1002 commented:
thank you
